oauth.go

```diff
@@ -56,8 +58,24 @@ func (r *oauthProxy) newOAuth2Config(redirectionURL string) *oauth2.Config {
 // NOTE: we may be able to extract the specific (non-standard) claim refresh_expires_in and refresh_expires
 // from response.RawBody.
 // When not available, keycloak provides us with the same (for now) expiry value for ID token.
-func getRefreshedToken(conf *oauth2.Config, t string) (jwt.JSONWebToken, string, string, time.Time, time.Duration, error) {
-	tkn, err := conf.TokenSource(context.Background(), &oauth2.Token{RefreshToken: t}).Token()
+func getRefreshedToken(conf *oauth2.Config, r *oauthProxy, t string) (jwt.JSONWebToken, string, string, time.Time, time.Duration, error) {
```

Comment:
If we only need the `r.config.OpenIDProviderTimeout` and `r.config.SkipOpenIDProviderTLSVerify`, we should just pass in those two values into this function. Or we could pass in the `config` struct. I don't think we should pass in the entire `oauthProxy` struct.